src/lib.rs

```diff
@@ -842,4 +843,12 @@ mod tests {
         let slice = bytes.as_slice_of::<u16>().unwrap();
         assert_eq!(slice, &[]);
     }
+
+    #[test]
+    fn unit() {
+        let slice: [(); 4] = [(), (), (), ()];
```

Comment:
For your use-case, how do you encode the length of the slice original slice if it always ends up with an empty byte slice?

Reply:
There is no need to encode the length in my use-case as `()` just represents that an edge has no value attached.